packages/application-components/src/components/dialogs/internals/dialog-container.tsx

```diff
@@ -27,18 +27,28 @@ const getDefaultParentSelector = () =>
         `#${PORTALS_CONTAINER_ID}`
       ) as HTMLElement);
 
+const getOverlayElement: ModalProps['overlayElement'] = (
+  props,
+  contentElement
+) => (
+  // Assign the `data-role` to the overlay container, which is used as
+  // the CSS selector in the `<PortalsContainer>`.
+  <div {...props} data-role="dialog-overlay">
+    {contentElement}
+  </div>
+);
+
 type Props = {
   isOpen: boolean;
   onClose?: (event: SyntheticEvent) => void;
   size: 'm' | 'l' | 7 | 8 | 9 | 10 | 'scale';
-  zIndex: number;
+  zIndex?: number;
```

Comment:
Are there actual use cases to set it explicitly given the automatic mechanisms implemented here? Should we deprecate it maybe?